internal/manager/conditions_ingressapi_test.go

```diff
@@ -0,0 +1,148 @@
+package manager_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	netv1 "k8s.io/api/networking/v1"
+	netv1beta1 "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+)
+
+type ingressTestOpts struct {
+	enabled      bool
+	crdInstalled bool
+}
+
+func TestIngressControllerConditions(t *testing.T) {
+	const kind = "Ingress"
+	var (
+		networkingV1 = schema.GroupVersionResource{
+			Group:    netv1.SchemeGroupVersion.Group,
+			Version:  netv1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+		networkingV1beta1 = schema.GroupVersionResource{
+			Group:    netv1beta1.SchemeGroupVersion.Group,
+			Version:  netv1beta1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+		extensionsV1beta1 = schema.GroupVersionResource{
+			Group:    extensionsv1beta1.SchemeGroupVersion.Group,
+			Version:  extensionsv1beta1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+	)
+
+	testCases := []struct {
+		name string
+
+		ingressNetV1      ingressTestOpts
+		ingressClassNetV1 ingressTestOpts
+		ingressNetV1beta  ingressTestOpts
+		ingressExtV1beta  ingressTestOpts
+
+		expectIngressNetV1      bool
+		expectIngressClassNetV1 bool
+		expectIngressNetV1beta  bool
+		expectIngressExtV1beta  bool
+		expectError             bool
+	}{
+		{
+			name:                    "netV1_takes_precedence_over_all",
+			ingressNetV1:            ingressTestOpts{true, true},
```

Comment:
Can we be explicit with fields assignment?

```suggestion
			ingressNetV1: ingressTestOpts{
				enabled:      true,
				crdInstalled: true,
			},
```

Reply:
My goal was to make this more compact, hence putting the definition in one line (IMO it's easier to compare visually test cases in this form).

I get your point on labels. I didn't use explicit ones because GoLand displays them so it didn't bother me. 
<img width="857" alt="image" src="https://user-images.githubusercontent.com/8835851/193785980-4adc58b4-0fc6-4263-8802-782b1cc4208e.png">

Are you ok with meeting between and leaving these definitions to be one-liners, but with explicit labels?
```
		{
			name:                    "netV1_takes_precedence_over_all",
			ingressNetV1:            ingressTestOpts{enabled: true, crdInstalled: true},
			ingressClassNetV1:       ingressTestOpts{enabled: true, crdInstalled: true},
			ingressNetV1beta:        ingressTestOpts{enabled: true, crdInstalled: true},
			ingressExtV1beta:        ingressTestOpts{enabled: true, crdInstalled: true},
			expectIngressNetV1:      true,
			expectIngressClassNetV1: true,
		},
```

Reply:
Looks fine for me 👍

Reply:
[5bc96d7](https://github.com/Kong/kubernetes-ingress-controller/pull/3013/commits/5bc96d7e375ec6dfb81b4e32710fede95a357911)